internal/store/memdb.go

```diff
@@ -39,6 +42,106 @@ func (memDb *MemDb) Load(tmcfFile string) error {
 	if err != nil {
 		return err
 	}
-	memDb.statSeries = map[string]map[string]pb.SeriesMap{}
+	memDb.statSeries = map[string]map[string][]*pb.Series{}
+	return nil
+}
+
+// nodeObs holds information for one observation
+type nodeObs struct {
+	statVar string
+	place   string
+	date    string // Unpaired date
+	value   string // Unpaired value
+	meta    *pb.StatMetadata
+}
+
+// addRow adds one csv row to memdb
+func addRow(
+	header []string,
+	row []string,
+	schemaMapping *parser.TableSchema,
+	statSeries map[string]map[string][]*pb.Series,
+) error {
+	// Keyed by node id like "E0"
+	allNodes := map[string]*nodeObs{}
+	// Initialize observation entries with the fixed schema
+	for node, meta := range schemaMapping.NodeSchema {
+		if typ, ok := meta["typeOf"]; ok && typ == "StatVarObservation" {
+			allNodes[node] = &nodeObs{
+				statVar: meta["variableMeasured"],
+				meta:    &pb.StatMetadata{},
+			}
+		}
+		if v, ok := meta["measurementMethod"]; ok {
+			allNodes[node].meta.MeasurementMethod = v
+		}
+		if v, ok := meta["unit"]; ok {
+			allNodes[node].meta.Unit = v
+		}
+		if v, ok := meta["scalingFactor"]; ok {
+			allNodes[node].meta.ScalingFactor = v
+		}
```

Comment:
These can be constants or can be in the column too? e.g., https://github.com/datacommonsorg/data/blob/master/scripts/un/energy/un_energy.tmcf#L8-L10